app/models/trigger_metadata_event_callback.rb

```diff
@@ -15,12 +18,14 @@ def self.publish
   def environment(model)
     if model.aggregate.respond_to?(:component)
       model.aggregate.component.environment
-    elsif model.respond_to?(:service) && model.respond_to?(:sp_component_id)
-      model.service.sp_component.environment
-    elsif model.respond_to?(:service) && model.respond_to?(:msa_component_id)
-      model.service.msa_component.environment
+    elsif model.respond_to?(:service)
+      if model.service.sp_component_id.present?
+        model.service.sp_component.environment
+      elsif model.service.msa_component_id.present?
```

Comment:
one last request 
```
  def environment(model)
    if model.aggregate.respond_to?(:component)
      model.aggregate.component.environment
    elsif model.respond_to?(:service)
      if model.service.sp_component_id.present?
        model.service.sp_component.environment
      elsif model.service.msa_component_id.present?
        model.service.msa_component.environment
      else
        nil
      end
    else
      model.aggregate.environment
    end
  end
```
It would make it easier to understand what going in future